js/redux/modules/filter.js

```diff
@@ -10,11 +23,41 @@ const UPDATE_OCCUPANT_TAGS = 'UPDATE_OCCUPANT_TAGS';
 const UPDATE_NUM_BATHROOMS = 'UPDATE_NUM_BATHROOMS';
 const UPDATE_NUM_BEDROOMS = 'UPDATE_NUM_BEDROOMS';
 const UPDATE_PRICE_RANGE = 'UPDATE_PRICE_RANGE';
+const UPDATE_SORT_OPTIONS = 'UPDATE_SORT_OPTIONS';
 
-export const updatePriceRange = priceRange => ({
-  type: UPDATE_PRICE_RANGE,
-  payload: priceRange
-});
+export const queryBasedOnFilters = async () => {
+  const {
+    priceRange,
+    propertyTags,
+    laundryTags,
+    location,
+    occupantTags,
+    parkingTags,
+    otherTags,
+    numBathrooms,
+    numBedrooms,
+    sortOptions
+  } = store.getState().filter;
+
+  const listingArray = await getNewListings(location);
+  let results = [];
+  listingArray.docs.forEach(result => results.push(unMarshallResult(result)));
```

Comment:
Line 43 and 44, I have seen this pattern a lot. You can use the map function to get the same result.

Reply:
done - good call :-)